src/vos/tests/parse_csv.py

```diff
@@ -0,0 +1,261 @@
+        yaml_file.write("""posix_key: &posix
+  objects: [
+{0}
+  ]
+
+containers: [*posix]
+""".format(split(obj, 4)))
+
+
+def ingest(fname, args):
+    """Parse csv and produce yaml input to vos_size.py"""
+    value_dict = {}
+    with open(fname, "r") as csv_file:
+        idx = 0
+        fields = csv_file.readline().strip().split(',')
+        values = csv_file.readline().strip().split(',')
+        if len(fields) != len(values):
+            print "CSV must provide one row of values that matches fields"
+            print "Number of fields is %d" % len(fields)
+            print "Number of values is %d" % len(values)
+            sys.exit(-1)
+        for name in fields:
+            value_dict[name] = values[idx]
+            idx += 1
+        file_histo = {}
+        for size in FILE_SIZES:
+            num_files = int(value_dict["%s_count" % size])
+            total_size = int(value_dict["%s_size" % size])
+            if num_files != 0:
+                file_histo[size] = {"count": num_files,
+                                    "avgsize": total_size / num_files}
+        dir_count = int(value_dict["dir_count"])
+        link_count = int(value_dict["link_count"])
+        if link_count < dir_count:
+            link_count = dir_count # assume at least one link per dir
+        obj_count = int(value_dict["total_objects"])
+        dir_histo = {"count": dir_count,
+                     "avgfiles": obj_count / dir_count,
+                     "avglink": link_count / dir_count}
+        produce_results(args, file_histo, dir_histo)
+
+def run_main():
+    """Run the program"""
+    import argparse
+
+    parser = argparse.ArgumentParser(description="Estimate VOS Overhead")
+    parser.add_argument('csv', metavar='CSV', type=str, nargs=1,
+                        help='Input CSV (assumes Argonne format)')
+    parser.add_argument('--name-len', type=int, dest="name",
+                        help='Average file name length', default=32)
+    parser.add_argument('--link-len', type=int, dest="link",
+                        help='Average link length', default=32)
+    parser.add_argument('--chunk-size', dest="chunk_size", type=int,
+                        help='Array chunk size.  Must be multiple of I/O size',
+                        default=1048576)
+    parser.add_argument('--num-pools', dest="num_pools", type=int,
+                        help='Number of vos pools', default=1000)
+    parser.add_argument('--io-size', dest="io_size", type=int,
+                        help='I/O size', default=1048576)
+    parser.add_argument('--output', dest="output", type=str,
+                        help='Output file name', default="output.yaml")
+    args = parser.parse_args()
+
+    if args.io_size > args.chunk_size:
+        print "Assuming --io-size set to --chunk-size %d\n" % args.chunk_size
+        args.io_size = args.chunk_size
+
+    if args.chunk_size % args.io_size != 0:
+        print "--chunk-size must be evenly divisble by --io-size"
```

Comment:
is this just a requirement for the tool to estimate properly?

Reply:
Yeah, I think it just makes the math a bit simpler that way.  Is it not generally the case?

Reply:
no doesn't have to be. applications can issue I/Os with any size regardless of the chunk size.
but for this tool and estimation, it's is an OK assumption.